src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -4188,12 +4188,14 @@ void emitter::emitIns_R_R(
             if (ins == INS_xtn)
             {
                 assert(size == EA_8BYTE);
+                // Element size can't be 8 bytes as single element vectors not
```

Comment:
Can you please explain why the assertion needs to be moved here?

As far as I understand, we are keeping opt and size semantics - i.e. they continue to correspond to Tb:
`XTN{2} <Vd>.<Tb>, <Vn>.<Ta>`

Tb can be 8B, 4H, 2S for xtn or 16B, 8H, 4S for xtn2. So this assertion should apply to both instructions, shouldn't it?

Reply:
If `size` and `elemsize` are both `8 bytes` that means you have a `64bit`  vector with a `64bit element` no? so that's e.g. `int64x1_t`. `8B` is `size == EA_8BYTE` and `elemsize == EA_BYTE`  (e.g. int8x8_t) unless I misunderstood?

Reply:
Yes, there is no question that this assert is valid for instruction xtn.

I was wondering why we do not `assert(elemsize != EA_8BYTE)` for xtn2 as we did before? `.2D` is not supported vector arrangement for xtn2, right?

Reply:
indeed. The assert was moved when it was using the input parameters as size. This is indeed not needed for the output parameter